src/donation/donation.controller.ts

```diff
@@ -1,12 +1,96 @@
-import { Controller, Get } from '@nestjs/common';
+import {
+  Body,
+  ClassSerializerInterceptor,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  Param,
+  ParseIntPipe,
+  Post,
+  UseInterceptors,
+} from '@nestjs/common';
 import { DonationService } from './donation.service';
+import { DonationEntity } from './models/donation.entity';
+import {
+  ApiBadRequestResponse,
+  ApiCreatedResponse,
+  ApiNoContentResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { CreateDonationDto } from './dto/create-donation.dto';
 
+@ApiTags('Donations Management')
+@UseInterceptors(ClassSerializerInterceptor)
 @Controller('donations')
 export class DonationController {
   constructor(private readonly donationService: DonationService) {}
 
   @Get()
-  hello(): string {
-    return this.donationService.hello();
+  @ApiOperation({
+    summary: 'Get Donations',
+    description: 'Returns a list of all donations.',
+  })
+  @ApiOkResponse({
+    type: DonationEntity,
+    isArray: true,
+    description: 'List of all added donations.',
+  })
+  getDonations(): Promise<DonationEntity[]> {
+    return this.donationService.findDonations();
+  }
```

Comment:
Actually true. I've added it when first setting up a simple CRUD and then expanded and it sorta just stayed there.
I can delete it if that's preferable.

Reply:
Having such a method would allow for someone to easily extract all of our users' data, if we left that endpoint without proper authentication. Considering we don't have a valid business use-case, I'd prefer it to be unavailable 😅 

If at any point we need to extract all donation data, we can always extract that directly from the database, without the need for programatic access via a REST API.

Reply:
Ok, I will remove it.